### ISSUE TITLE
Add possibility to disable checking of whole source classes

### ISSUE THREAD

**Issue by psiroky** [ImmobilienScout24/illegal-transitive-dependency-check#29]
I know there is a option `regexIgnoredClass` that let's me specify classes that should be ignored when they are being used, but are not declared as direct dependency.

I think it would be great to also have a possibility to configure classes, that should be excluded from the check itself. So if they use some transitive dependency, it would be just ignored. 

What do you think?

**Comment by 2bad4u** [comment]
Hi Petr,

the filter created from `regexIgnoredClass` property is also applied to the classes loaded from the artifact beeing under examination and thus their dependencies won't be counted. Therefore I suppose it to work the way you want if you specify your classes in `regexIgnoredClass`. Can you give it a try please?

Regards
André

**Comment by psiroky** [comment]
Hmm, this is interesting. I tried that, but the classes did not get excluded. Let me try again, I will let you know.

**Comment by psiroky** [comment]
Hi André,

sorry, I am not sure I am explaining this correctly. 

What I would like to do is to disable the checking of the contents of my own class. For example: I have a class called `MyBadClass` that transitively depends on `slf4j-api` classes. This causes the build to fail, because I am not declaring `slf4j` as direct dependency.
Now I would like to configure the rule to ignore that class altogether, so that the build would pass. (`<regexIgnoredClass>.*MyBadClass.*\</regexIgnoreClass\>`). I could not make this work with version `1.7.3`. Is this something that should be already doable? 

I'll post an example maven project which demonstrates what I would like to achieve.

Thanks for the quick response!
Petr

**Comment by psiroky** [comment]
This is a simple example that demonstrates my intent: https://github.com/psiroky/playground/tree/master/transitive-dep-check-example.

**Comment by 2bad4u** [comment]
Thanks for providing the example! I've completely understood your request but my expectation has been wrong (I simply didn't look into the code for a long time :)). I'll make the necessary changes in the next days and release a new version..

**Comment by psiroky** [comment]
Thanks André!
